Add a list of external PyGMT resources

seisman opened the pull request:
**Description of proposed changes**



<!-- Please describe changes proposed and **why** you made them. If unsure, open an issue first so we can discuss.-->

<!-- If fixing an issue, put the issue number after the # below (no spaces). GitHub will automatically close it when this gets merged. -->
Fixes #1059


**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

meghanrjones commented:
Thanks for submitting this PR - the resources are great! 

I would prefer that these resources go under a different section than 'getting started'. One option would be to have a new "Examples" section that could include the "Gallery" and "External examples". I think that the external tutorials could be included in that section for now, but long term could go into the redesigned tutorial sidebar (https://github.com/GenericMappingTools/pygmt/issues/1191). 

Also, I prefer "External examples" or something else over "User Contributed Resources" to make it more clear PyGMT is not trying to take ownership of these resources and is just providing links. I think this could make people who aren't PyGMT developers more likely to include links to their PyGMT projects in the page.

michaelgrund commented:
> 
> 
> Thanks for submitting this PR - the resources are great!
> 
> I would prefer that these resources go under a different section than 'getting started'. One option would be to have a new "Examples" section that could include the "Gallery" and "External examples". I think that the external tutorials could be included in that section for now, but long term could go into the redesigned tutorial sidebar (#1191).
> 
> Also, I prefer "External examples" or something else over "User Contributed Resources" to make it more clear PyGMT is not trying to take ownership of these resources and is just providing links. I think this could make people who aren't PyGMT developers more likely to include links to their PyGMT projects in the page.

I agree with @meghanrjones , would also use something like "External examples".

seisman commented:
> I would prefer that these resources go under a different section than 'getting started'. One option would be to have a new "Examples" section that could include the "Gallery" and "External examples". I think that the external tutorials could be included in that section for now, but long term could go into the redesigned tutorial sidebar (#1191).
> 
> Also, I prefer "External examples" or something else over "User Contributed Resources" to make it more clear PyGMT is not trying to take ownership of these resources and is just providing links. I think this could make people who aren't PyGMT developers more likely to include links to their PyGMT projects in the page.

Good suggestion. I renamed the page to "External Resources" and moved it to the "Examples" section.

meghanrjones commented on doc/external_resources.md:
```diff
@@ -0,0 +1,20 @@
+
+Below is a curated collection of external PyGMT resources.
+
+*To add your contribution to this collection, submit a recommendation via a
+[pull request](https://github.com/GenericMappingTools/pygmt/blob/master/doc/external_resources.md).*
```
I think adding to the documentation editing instructions would be helpful here and am open to your preferred wording.
```suggestion
*To add your contribution to this collection, follow [these instructions](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#editing-the-documentation)
to submit a pull request with your recommended addition to the
[External Resources file](https://github.com/GenericMappingTools/pygmt/blob/master/doc/external_resources.md).*
```

meghanrjones reviewed (commented):
Looks good to me! I just have one small comment about the instructions for adding an example.

meghanrjones reviewed (approved):
Looks good to me.

meghanrjones commented on doc/external_resources.md:
```diff
@@ -0,0 +1,21 @@
+- [2021 PyGMT course at the UAF Geophysical Institute](https://github.com/liamtoney/gi-pygmt-2021) by Liam Toney
+- [Remote Online Sessions for Emerging Seismologists (ROSES): Unit 8 - PyGMT](https://www.iris.edu/hq/inclass/lesson/728) by Liam Toney
+- [PyGMT Tutorial in 2021](https://github.com/MIGG-NTU/PyGMT2021) by MIGG-NTU
+- [PyGMT Workshop at FOSS4G Oceania 2019](https://github.com/GenericMappingTools/foss4g2019oceania) by Wei Ji
```
Before merging, please confirm whether @weiji14 prefers this or full name format here and in L21.

weiji14 replied:
Prefer to include my surname here :)

```suggestion
- [PyGMT Workshop at FOSS4G Oceania 2019](https://github.com/GenericMappingTools/foss4g2019oceania) by Wei Ji Leong
```

core-man commented:
Looks great~

michaelgrund reviewed (approved):
Looks good!

seisman commented on doc/external_resources.md:
```diff
@@ -0,0 +1,21 @@
+
+- [2021 PyGMT course at the UAF Geophysical Institute](https://github.com/liamtoney/gi-pygmt-2021) by Liam Toney
+- [Remote Online Sessions for Emerging Seismologists (ROSES): Unit 8 - PyGMT](https://www.iris.edu/hq/inclass/lesson/728) by Liam Toney
+- [PyGMT Tutorial in 2021](https://github.com/MIGG-NTU/PyGMT2021) by MIGG-NTU
```
@core-man Are you OK with "MIGG-NTU"? Or should your name be listed here, since you're the only contributor of the https://github.com/MIGG-NTU/PyGMT2021 tutorial?

core-man replied:
Please use MIGG-NTU.

seisman commented:
Ping @liamtoney and @weiji14 as their contributions are listed.

weiji14 commented on doc/external_resources.md:
```diff
@@ -0,0 +1,21 @@
+## Examples
+
+- [GMT and PyGMT plotting examples](https://github.com/michaelgrund/GMT-plotting) by Michael Grund
+- [NZ Antarctic Science Conference 2021 poster](https://github.com/weiji14/nzasc2021) by Wei Ji
```
```suggestion
- [NZ Antarctic Science Conference 2021 poster](https://github.com/weiji14/nzasc2021) by Wei Ji Leong
```

weiji14 commented on doc/external_resources.md:
```diff
@@ -0,0 +1,21 @@
+- [PyGMT Tutorial in 2021](https://github.com/MIGG-NTU/PyGMT2021) by MIGG-NTU
+- [PyGMT Workshop at FOSS4G Oceania 2019](https://github.com/GenericMappingTools/foss4g2019oceania) by Wei Ji
+
+## Examples
```
Could we change this to be a bit more descriptive? Something like "Examples from Publications and Posters", or something along those lines?

seisman replied:
Done in 1a1670759b4a48b49558d93c5052cdcf7e326b80.
